addons/l10n_in_tds_tcs_warning/models/account_move.py

```diff
@@ -0,0 +1,108 @@
+from odoo import api, fields, models
+from odoo.tools import SQL
+from odoo.tools.date_utils import get_month
+
+
+class AccountMove(models.Model):
+    _inherit = "account.move"
+
+    l10n_in_tcs_tds_warning = fields.Char('TDC/TCS Warning', compute="_compute_l10n_in_tcs_tds_warning", store=True)
+
+    def _get_sections_aggregate_sum_by_pan(self, section_alert, commercial_partner_id):
+        self.ensure_one()
+        month_start_date, month_end_date = get_month(self.date)
+        company_fiscalyear_dates = self.company_id.compute_fiscalyear_dates(self.date)
+        fiscalyear_start_date, fiscalyear_end_date = company_fiscalyear_dates['date_from'], company_fiscalyear_dates['date_to']
+        default_domain = [
+            ('account_id.l10n_in_tds_tcs_section', '=', section_alert.id),
+            ('move_id.move_type', '!=', 'entry'),
+            ('company_id', 'child_of', self.company_id.root_id.id),
+        ]
+        if commercial_partner_id.l10n_in_pan:
+            default_domain += [('move_id.commercial_partner_id.l10n_in_pan', '=', commercial_partner_id.l10n_in_pan)]
+        else:
+            default_domain += [('move_id.commercial_partner_id', '=', commercial_partner_id.id)]
+        frequency_domains = {
+            'monthly': [('date', '>=', month_start_date), ('date', '<=', month_end_date)],
+            'fiscal_yearly': [('date', '>=', fiscalyear_start_date), ('date', '<=', fiscalyear_end_date)],
+        }
+        aggregate_result = {}
+        for frequency, frequency_domain in frequency_domains.items():
+            query = self.env['account.move.line']._where_calc(default_domain + frequency_domain)
+            result = self.env.execute_query_dict(SQL(
+                """
+                SELECT sum(account_move_line.balance) as balance,
+                       sum(account_move_line.price_total * am.invoice_currency_rate) as price_total
+                  FROM %s
+                  JOIN account_move AS am ON am.id = account_move_line.move_id
+                 WHERE %s
+                """,
+                query.from_clause,
+                query.where_clause)
```

Comment:
Let's not store this field (l10n_in_tcs_tds_warning) and only compute it on `invoice_line_ids.price_subtotal` is changed so it will be applicable to form view, still we can change the query if you want